packages/parameters/tests/e2e/appConfigProvider.class.test.ts

```diff
@@ -0,0 +1,355 @@
+ * 
+ * The parameters created are:
+ * - Free-form JSON
+ * - Free-form YAML
+ * - 2x Free-form plain text
+ * - Feature flag
+ * 
+ * These parameters allow to retrieve the values and test some transformations.
+ * 
+ * The tests are:
+ * 
+ * Test 1
+ * get a single parameter as-is (no transformation)
+ * 
+ * Test 2
+ * get a free-form JSON and apply binary transformation (should return a stringified JSON)
+ * 
+ * Test 3
+ * get a free-form YAML and apply binary transformation (should return a string-encoded YAML)
+ * 
+ * Test 4
+ * get a free-form plain text and apply binary transformation (should return a string)
+ * 
+ * Test 5
+ * get a feature flag and apply binary transformation (should return a stringified JSON)
+ * 
+ * Test 6
+ * get parameter twice with middleware, which counts the number of requests, 
+ * we check later if we only called AppConfig API once
+ * 
+ * Test 7
+ * get parameter twice, but force fetch 2nd time, we count number of SDK requests and
+ * check that we made two API calls
+ */
+describe(`parameters E2E tests (appConfigProvider) for runtime ${runtime}`, () => {
+
+  let invocationLogs: InvocationLogs[];
+  const encoder = new TextEncoder();
+
+  beforeAll(async () => {
+    // Create a stack with a Lambda function
+    stack = createStackWithLambdaFunction({
+      app: integTestApp,
+      stackName,
+      functionName,
+      functionEntry: path.join(__dirname, lambdaFunctionCodeFile),
+      environment: {
+        UUID: uuid,
+
+        // Values(s) to be used by Parameters in the Lambda function
+        APPLICATION_NAME: applicationName,
+        ENVIRONMENT_NAME: environmentName,
+        FREEFORM_JSON_NAME: freeFormJsonName,
+        FREEFORM_YAML_NAME: freeFormYamlName,
+        FREEFORM_PLAIN_TEXT_NAME_A: freeFormPlainTextNameA,
+        FREEFORM_PLAIN_TEXT_NAME_B: freeFormPlainTextNameB,
+        FEATURE_FLAG_NAME: featureFlagName,
+      },
+      runtime,
+    });
+
+    // Create the base resources for an AppConfig application.
+    const {
+      application,
+      environment,
+      deploymentStrategy
+    } = createBaseAppConfigResources({
+      stack,
+      applicationName,
+      environmentName,
+      deploymentStrategyName,
+    });
+
+    // Create configuration profiles for tests.
+    const freeFormJson = createAppConfigConfigurationProfile({
+      stack,
+      application,
+      environment,
+      deploymentStrategy,
+      name: freeFormJsonName,
+      type: 'AWS.Freeform',
+      content: {
+        content: JSON.stringify(freeFormJsonValue),
+        contentType: 'application/json',
+      }
+    });
+
+    const freeFormYaml = createAppConfigConfigurationProfile({
+      stack,
+      application,
+      environment,
+      deploymentStrategy,
+      name: freeFormYamlName,
+      type: 'AWS.Freeform',
+      content: {
+        content: freeFormYamlValue,
+        contentType: 'application/x-yaml',
+      }
+    });
+    freeFormYaml.node.addDependency(freeFormJson);
```

Comment:
~I removed this chained dependencies and could not reproduce the described behaviour. Can you rerun and double check that the previous test run stack does not exist?~

This is because all deployment are running within the same app. Running one deployment per application is possible, we can spawn new app for each `createAppConfigConfigurationProfile` call. What do you think?

Reply:
We could do that yes, however we would move the "ugly" part to the function code of the test instead because we would have to create a dedicated instance of AppConfigProvider for each application.

Right now creating a provider requires to do this:
```ts
const provider = new AppConfigProvider({ application: 'something', environment: 'something' });
```

So if we have multiple environments/applications, we'll need a provider for each pair.

What do you think?

Reply:
I have moved the `environment` to the inner part of the resource creation and 10 tests were green, but I was only lucky, the race condition remains (tested today again). Let's keep the workaround with a comment.

Reply:
Done, I have added a short paragraph in the comment section of this test to explain & document this.